deflect/SizeHints.h

```diff
@@ -73,6 +73,10 @@ struct SizeHints
     //@}
 };
 
+#if _MSC_VER
+#  define noexcept
```

Comment:
I was building with ninja, and it only reports _MSC_VER w/o any version iirc. The proper fix would be to add it to CMake/common, like we did for override and final.